diff -Nur Mail-SpamAssassin-3.4.1.orig/spamc/configure Mail-SpamAssassin-3.4.1/spamc/configure
--- Mail-SpamAssassin-3.4.1.orig/spamc/configure	2015-04-28 13:56:59.000000000 -0600
+++ Mail-SpamAssassin-3.4.1/spamc/configure	2016-11-06 14:36:54.722342568 -0700
@@ -3666,9 +3666,9 @@
 SSLLIBS=""
 SSLCFLAGS=""
 if test yes = "$sa_ssl_enabled"; then
-	echo "$as_me:$LINENO: checking for CRYPTO_lock in -lcrypto" >&5
-echo $ECHO_N "checking for CRYPTO_lock in -lcrypto... $ECHO_C" >&6
-if test "${ac_cv_lib_crypto_CRYPTO_lock+set}" = set; then
+	echo "$as_me:$LINENO: checking for CRYPTO_malloc in -lcrypto" >&5
+echo $ECHO_N "checking for CRYPTO_malloc in -lcrypto... $ECHO_C" >&6
+if test "${ac_cv_lib_crypto_CRYPTO_malloc+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
@@ -3686,11 +3686,11 @@
 #endif
 /* We use char because int might match the return type of a gcc2
    builtin and then its argument prototype would still apply.  */
-char CRYPTO_lock ();
+char CRYPTO_malloc ();
 int
 main ()
 {
-CRYPTO_lock ();
+CRYPTO_malloc ();
   ;
   return 0;
 }
@@ -3716,20 +3716,20 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  ac_cv_lib_crypto_CRYPTO_lock=yes
+  ac_cv_lib_crypto_CRYPTO_malloc=yes
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
 
-ac_cv_lib_crypto_CRYPTO_lock=no
+ac_cv_lib_crypto_CRYPTO_malloc=no
 fi
 rm -f conftest.err conftest.$ac_objext \
       conftest$ac_exeext conftest.$ac_ext
 LIBS=$ac_check_lib_save_LIBS
 fi
-echo "$as_me:$LINENO: result: $ac_cv_lib_crypto_CRYPTO_lock" >&5
-echo "${ECHO_T}$ac_cv_lib_crypto_CRYPTO_lock" >&6
-if test $ac_cv_lib_crypto_CRYPTO_lock = yes; then
+echo "$as_me:$LINENO: result: $ac_cv_lib_crypto_CRYPTO_malloc" >&5
+echo "${ECHO_T}$ac_cv_lib_crypto_CRYPTO_malloc" >&6
+if test $ac_cv_lib_crypto_CRYPTO_malloc = yes; then
   SSLLIBS="-lcrypto $SSLLIBS"
 fi
 
@@ -3804,7 +3804,7 @@
 	# before defining SPAMC_SSL check that all its requirements are
 	# actually available
 	if test yes = "$ac_cv_header_openssl_crypto_h" && \
-	   test yes = "$ac_cv_lib_crypto_CRYPTO_lock" && \
+	   test yes = "$ac_cv_lib_crypto_CRYPTO_malloc" && \
 	   test yes = "$ac_cv_lib_ssl_SSL_CTX_free"; then
 		SSLCFLAGS="-DSPAMC_SSL"
 	else
diff -Nur Mail-SpamAssassin-3.4.1.orig/spamc/configure.in Mail-SpamAssassin-3.4.1/spamc/configure.in
--- Mail-SpamAssassin-3.4.1.orig/spamc/configure.in	2015-04-28 13:56:59.000000000 -0600
+++ Mail-SpamAssassin-3.4.1/spamc/configure.in	2016-11-06 14:36:54.724342587 -0700
@@ -64,13 +64,13 @@
 SSLLIBS=""
 SSLCFLAGS=""
 if test yes = "$sa_ssl_enabled"; then
-	AC_CHECK_LIB(crypto, CRYPTO_lock,[SSLLIBS="-lcrypto $SSLLIBS"])
+	AC_CHECK_LIB(crypto, CRYPTO_malloc,[SSLLIBS="-lcrypto $SSLLIBS"])
 	AC_CHECK_LIB(ssl, SSL_CTX_free,[SSLLIBS="-lssl $SSLLIBS"],,-lcrypto)
 
 	# before defining SPAMC_SSL check that all its requirements are
 	# actually available
 	if test yes = "$ac_cv_header_openssl_crypto_h" && \
-	   test yes = "$ac_cv_lib_crypto_CRYPTO_lock" && \
+	   test yes = "$ac_cv_lib_crypto_CRYPTO_malloc" && \
 	   test yes = "$ac_cv_lib_ssl_SSL_CTX_free"; then
 		SSLCFLAGS="-DSPAMC_SSL"
 	else
