--- Mail-SpamAssassin-3.4.1-orig/spamd/spamd.raw	2017-10-12 19:22:00.908132633 -0400
+++ Mail-SpamAssassin-3.4.1/spamd/spamd.raw	2017-06-25 15:21:38.627287220 -0400
@@ -1390,6 +1390,7 @@
 
     # handle $clients_per_child connections, then die in "old" age...
     my $orders;
+    opendir (my $orig_root, "/");
     for ( my $i = 0 ; $i < $clients_per_child ; $i++ ) {
       if ($scaling) {
         $scaling->update_child_status_idle();
@@ -1426,6 +1427,8 @@
           die("spamd: return setuid failed");
         }
       }
+      chdir($orig_root);
+      chroot(".");
 
       if ($copy_config_p) {
         # use a timeout!  There are bugs in Storable on certain platforms
@@ -2602,6 +2605,9 @@
   if ($dir) {
     my $cf_file = $dir . "/.spamassassin/user_prefs";
     create_default_cf_if_needed( $cf_file, $username, $dir );
+    if ( -f "/etc/mail/spamassassin/prefs" ) {
+        $spamtest->read_scoreonly_config('/etc/mail/spamassassin/prefs');
+    }
     $spamtest->read_scoreonly_config($cf_file);
   }
 
