diff -up Mail-SpamAssassin-3.4.1/spamc/configure.pl.orig Mail-SpamAssassin-3.4.1/spamc/configure.pl
--- Mail-SpamAssassin-3.4.1/spamc/configure.pl.orig	2017-05-18 13:53:34.340378346 +0200
+++ Mail-SpamAssassin-3.4.1/spamc/configure.pl	2017-05-18 13:52:55.950537948 +0200
@@ -66,7 +66,7 @@ print join(' ', $Config{'perlpath'}, "ve
   # Do the same thing as for the preprocessor below.
   package version_h;
   my $Z = $0;
-  local $0    = "version.h.pl";
+  local $0    = "./version.h.pl";
   local @ARGV = ();
   # Got to check for defined because the script returns shell error level!
   unless (defined do $0) {
